data/src/main/java/org/apache/iceberg/data/DeleteFilter.java

```diff
@@ -94,6 +96,23 @@ protected DeleteFilter(String filePath, List<DeleteFile> deletes, Schema tableSc
     this.eqDeletes = eqDeleteBuilder.build();
     this.requiredSchema = fileProjection(tableSchema, requestedSchema, posDeletes, eqDeletes);
     this.posAccessor = requiredSchema.accessorForField(MetadataColumns.ROW_POSITION.fieldId());
+    this.hasColumnIsDeleted = requestedSchema.findField(MetadataColumns.IS_DELETED.fieldId()) != null;
+    this.columnIsDeletedIndex = getIndexOfIsDeletedMetadataColumn();
+  }
+
+  private int getIndexOfIsDeletedMetadataColumn() {
+    List<Types.NestedField> icebergFields = requiredSchema.asStruct().fields();
+    for (int i = 0; i < icebergFields.size(); i++) {
+      if (icebergFields.get(i).fieldId() == MetadataColumns.IS_DELETED.fieldId()) {
+        return i;
+      }
+    }
+
+    return -1;
```

Comment:
[question] should we add this helper (which now takes fieldId and returns it's column position) in [Schema](https://github.com/apache/iceberg/blob/master/api/src/main/java/org/apache/iceberg/Schema.java), Your thoughts ?

Reply:
We should, thanks for the suggestion.